docs/virtual_machine_template_guide.adoc

```diff
@@ -0,0 +1,189 @@
+❽ PersistenVolumeClaim’s name. +
+
+=== Labels
+
+Labels will be presented on every OCP template object in order for ManageIQ to recognize it as a KubeVirt template.
+The labels specified in the template are applied to every object that is generated from the template.
+
+[source,yaml]
+----
+labels:
+  # vm-template label applied to all objects created from this template
+  template: “miq.github.io/kubvirt/vm-template”
+----
+
+There is also the ability to add labels in the template from the command line:
+
+----
+$ oc process -f <filename> -l name=“miq.github.io/kubvirt/vm-template”
+----
+
+=== Parameters
+
+Parameters allow a value to be supplied by the user or generated when the template is instantiated. +
+Then, that value is substituted wherever the parameter is referenced. +
+The following parameters will be used for the offlineVirtualMachine name (and as a part of the persistent volume claim names) +
+Default values will be provided for memory and cpu, which will be used if the user does not supply a different value.
+
+[source,yaml]
+----
+parameters:
+- description: Name for the new VM     
+  name: NAME   
+- name: MEMORY
+  description: Amount of memory in MB
+  value: 512Mi  
+- name: CPU
+  description: Amount of cores
+  value: 4    
+----
+
+=== Creating VirtualMachineTemplate Objects using the Web Console
+TBD  (add screenshots)
+
+=== Generating the VirtualMachineTemplate Objects from CLI 
+The list of parameters that you can override are listed under  Parameters.
+You can list them with the CLI by using the following command and specifying the file to be used:
+----
+  $ oc process --parameters -f <filename>
+----
+
+Or create objects from a template by processing the template and piping the output to oc create:
+
+----
+  $ oc process --parameters -f <filename> | oc create -f -
+----
+
+
+=== Full VirtualMachine Template example
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha1
+kind: Template 
+metadata:
+  name: RHEL-7  #This will define the template’s name
+  # name: Microsoft-Windows-2012r1
+objects:
+- apiVersion: kubevirt.io/v1alpha1
+  kind: offlineVirtualMachine
+  metadata:
+    name: rhel-7-${NAME}
+  spec:
+    template:
+      metadata:
+      labels:
+        my: label
+      spec:
+        domain:
+          resources:
+            memory: ${MEMORY}
+            cpu: ${CPU}
+          devices:
+      disks:
+        name: disk0
+        volumeName: mypvc 
+      volumes:
+         name: mypvc
+         presistentVolumeClaim:
+      claimName: rhel-7-pvc-${NAME}
+- apiVersion: kubevirt.io/v1alpha1
+  kind: presistentVolumeClaim
+  metadata:
+    name: rhel-7-pvc-${NAME}
+labels:
+  # vm-template label applied to all objects created from this template
+  template: “miq.github.io/kubvirt/vm-template”           
+parameters:
+- description: Name for the new VM     
+  name: NAME   
+- description: Amount of memory in MB
+  name: MEMORY
+ 
+  value: 512Mi  #default
+- description: Amount of cores
+  name: CPU
```

Comment:
`cpu` in the `resources.requests` section refers to how much resources you want to assign to the vm (cgroup related like in https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/). `spec.domain.cpu.cores` specifies the number of cpu cores in the vm (see https://github.com/kubevirt/kubevirt/pull/638).

I wonder what the cpu template variable in the other virtualization providers in ManageIQ refers to. I would expect that people are more interested in dynamically specifying the resource allocation than the actual number of cpu cores. @jhernand maybe you know more about that?

Reply:
[Here](https://pemcg.gitbooks.io/introduction-to-cloudforms-automation/content/chapter15/provisioning_dialog.html) you can see how hardware section looks like in vm provisioning dialog.

Reply:
/cc @fromanirh